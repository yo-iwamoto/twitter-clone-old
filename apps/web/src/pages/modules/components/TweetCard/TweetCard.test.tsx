import { TweetCard } from '.';
import { render } from '@testing-library/react';

describe('TweetCard', () => {
  it('正しく表示されること', () => {
    // act
    const { getByText, asFragment } = render(
      <TweetCard
        tweet={{
          id: '1',
          text: 'Hello World!',
          createdAt: '2022-02-04T10:40:23.885Z',
        }}
      />
    );

    // assert
    expect(getByText('Hello World!')).toBeDefined();
    expect(getByText('2022-02-04T10:40:23.885Z')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });
});
