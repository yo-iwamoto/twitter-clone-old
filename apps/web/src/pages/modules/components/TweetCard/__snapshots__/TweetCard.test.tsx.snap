// Vitest Snapshot v1

exports[`TweetCard > 正しく表示されること 1`] = `
<DocumentFragment>
  <article
    class="whitespace-pre-wrap border-b p-4"
  >
    <p>
      Hello World!
    </p>
    <p
      class="text-xs text-gray-500"
    >
      2022-02-04T10:40:23.885Z
    </p>
  </article>
</DocumentFragment>
`;
